plugins/wasp/general-wasp-knowledge.md

```diff
@@ -0,0 +1,126 @@
+
+### Implementation Checklist
+
+1. Fetch https://wasp.sh/llms.txt to verify current syntax (see "Documentation" above)
+2. Check if the feature exists in the current codebase or starter template for reference patterns
+3. Identify which files need changes: wasp config file (`main.wasp` or `main.wasp.ts`), `schema.prisma`, `src/`
+4. Plan the order: schema → migrate → wasp config file → TypeScript
+
+### Schema Changes
+
+After ANY change to `schema.prisma`:
+1. Stop the dev server if running
+2. Run `wasp db migrate-dev --name <descriptive-name>`
+3. Restart the dev server
+
+## Project Reference
+
+### Structure
+
+```
+.
+├── .wasp/                    # Wasp output (auto-generated, do not edit)
+├── public/                   # Static assets
+├── src/                      # Feature code: server `operations.ts` and client `pages.tsx` files
+├── main.wasp or main.wasp.ts # Wasp config file: routes, pages, auth, operations, jobs, etc.
+├── schema.prisma             # Database schema (Prisma)
+```
+
+### Code Organization
+
+Organize by **feature** (vertical), not by type (client/server):
+
+```
+src/
+├── tasks/
+│   ├── TasksPage.tsx      # Page component
+│   ├── TaskList.tsx       # Component
+│   └── operations.ts      # Queries & actions
+├── auth/
+│   ├── LoginPage.tsx
+│   └── google.ts
+```
+
+### Starter Templates
+
+Highly recommend that the user chose one of the following templates when scaffolding a new Wasp app (this helps Claude understand Wasp's features and patterns):
+```bash
+wasp new my-basic-app -t basic # creates a basic starter app with core Wasp features like auth, operations, pages, etc.
+wasp new my-saas-app -t saas # creates a full-featured SaaS starter app with auth, payments, demo app, AWS S3, and more (OpenSaaS.sh)
+```
+
+See the **Starter Templates** section in https://wasp.sh/llms.txt for more templates.
+
+### Customization
+
+Wasp provides layers of customization on top of the tools it uses (vite, expressjs, react-query, client and server setup, etc.): see the **Project Setup & Customization** section in https://wasp.sh/llms.txt.
+
+### Advanced Features
+
+Wasp provides **advanced features**:
+- custom HTTP API endpoints
+- background (cron) jobs
+- type-safe links
+- websockets
+- middleware
+- email sending
+
+See the **Advanced Features** section in https://wasp.sh/llms.txt for more details.
+
+### Import Conventions
+
+**In TypeScript files:**
+- ✅ `import type { User } from 'wasp/entities'`
+- ✅ `import type { GetTasks } from 'wasp/server/operations'`
+- ✅ `import { getTasks, createTask, useQuery } from 'wasp/client/operations'`
+- ✅ `import { SubscriptionStatus } from '@prisma/client'` (for Prisma enums)
+- ✅ Local code: relative paths `import { X } from './X'`
+- ⚠️ Call actions directly using `async/await`. DO NOT use Wasp's `useAction` hook unless optimistic updates are needed.
+
+**In main.wasp:**
+- ✅ `fn: import { getTasks } from "@src/tasks/operations.ts"`
```

Comment:
I think we also recommend maybe not using .ts extensions? Or do we? I can't remember, but I would confirm it by checking the example apps and following what they do.